### ISSUE TITLE
Verify Tournament Selection/Ranking for Refiner + Discriminator

### ISSUE THREAD

**Issue by roddtalebi** [ezCGP/ezCGP#217]
Jason just wants to make sure that Anthony implemented everything correctly.
Implemented in individual_evaluate.py but defined in codes/utilities/gan_tournament_selection.py
...this is how we pick the best refiner and the best discriminator from all the saved checkpoints along the training process
```
        refiner_ratings, discriminator_ratings = get_graph_ratings(refiners,
                                                                   discriminators,
                                                                   validating_datalist[0],
                                                                   train_config['device'])
        best_refiner = refiners[refiner_ratings['r'].argmax()]
        best_discriminator = discriminators[discriminator_ratings['r'].argmax()]
```

Then also implemented in problem_simgan.py for objective_functions() so that we could rank the population.
```
        # Run tournament and add ratings
        if len(refiners) > 0:
            refiner_ratings, _ = get_graph_ratings(refiners, discriminators, self.validating_datalist[0], 'cpu')
```

**Comment by roddtalebi** [comment]
found an error in calc_acc() ...code was doing a max(1) on the softmax instead of an argmax(1) to get the label

**Comment by roddtalebi** [comment]
Question!
In the main loop, why are we running ```matches_per_pairing``` number of matches instead of just 1 match but with, proportionally, more samples per match? ...like its just running data through the refiner and then discriminator to get a score and then come up with some metric for who 'won' that match.

...maybe that's it...we just need multiple matches played to have a sampling of winnings+losses for the pairings. But if that's the case, why even map the score to a boolean win/loss? why not just keep the score as a [0,1] float? ...it's more accurate than averaging the number of match win/losses. But then again, I didn't read the paper.

**Comment by roddtalebi** [comment]
need to verify calculate_new_glicko_scores() but rest looks good
